packages/pigment-css-react/src/Grid.jsx

```diff
@@ -0,0 +1,311 @@
+        return '0 0 auto';
+      }
+
+      return '0 1 auto';
+    },
+    '--Grid-self-margin-left': (value) => {
+      if (value === 'auto') {
+        return 'auto';
+      }
+
+      return 'var(--Grid-fixed-offset)';
+    },
+  },
+}));
+
+const GridComponent = styled('div')({
+  '--Grid-fixed-width':
+    'calc(100% * var(--Grid-self-column-span) / var(--Grid-parent-column-count) - (var(--Grid-parent-column-count) - var(--Grid-self-column-span)) * var(--Grid-parent-column-spacing) / var(--Grid-parent-column-count))',
+  '--Grid-fixed-offset':
+    'calc(100% * var(--Grid-self-offset) / var(--Grid-parent-column-count) + var(--Grid-parent-column-spacing) * var(--Grid-self-offset) / var(--Grid-parent-column-count))',
+  variants: [
+    {
+      props: { container: true },
+      style: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: 'var(--Grid-self-row-spacing) var(--Grid-self-column-spacing)',
+      },
+    },
+    {
+      props: ({ size }) => size !== undefined,
+      style: {
+        width: 'var(--Grid-self-width)',
+        maxWidth: 'var(--Grid-self-max-width)',
+        flex: 'var(--Grid-self-flex)',
+      },
+    },
+    {
+      props: ({ offset }) => offset !== undefined,
+      style: {
+        marginLeft: 'var(--Grid-self-margin-left)',
+      },
+    },
+  ],
+});
+
+const Grid = React.forwardRef(function Grid(
+  {
+    children,
+    columns,
+    spacing,
+    columnSpacing,
+    rowSpacing,
+    direction = 'row',
+    style,
+    className,
+    component = 'div',
+    container = false,
+    size,
+    offset,
+    // internal props
+    // eslint-disable-next-line react/prop-types
+    unstable_parent_columns,
+    // eslint-disable-next-line react/prop-types
+    unstable_parent_column_spacing,
+    // eslint-disable-next-line react/prop-types
+    unstable_parent_row_spacing,
+    ...rest
+  },
+  ref,
+) {
+  const selfColumns = columns ?? unstable_parent_columns ?? 12;
+  const selfColumnSpacing = columnSpacing ?? spacing ?? unstable_parent_column_spacing ?? 0;
+  const selfRowSpacing = rowSpacing ?? spacing ?? unstable_parent_row_spacing ?? 0;
+
+  const GridAtomicsObj = {
+    direction,
+  };
+
+  if (unstable_parent_columns !== undefined) {
+    GridAtomicsObj['--Grid-parent-column-count'] = unstable_parent_columns;
+  }
+
+  if (unstable_parent_column_spacing !== undefined) {
+    GridAtomicsObj['--Grid-parent-column-spacing'] = unstable_parent_column_spacing;
+  }
+
+  if (unstable_parent_row_spacing !== undefined) {
+    GridAtomicsObj['--Grid-parent-row-spacing'] = unstable_parent_row_spacing;
+  }
+
+  if (container) {
+    GridAtomicsObj['--Grid-self-column-spacing'] = selfColumnSpacing;
+    GridAtomicsObj['--Grid-self-row-spacing'] = selfRowSpacing;
+  }
+
+  if (size) {
+    GridAtomicsObj['--Grid-self-column-span'] = size;
+    GridAtomicsObj['--Grid-self-width'] = size;
+    GridAtomicsObj['--Grid-self-max-width'] = size;
+    GridAtomicsObj['--Grid-self-flex'] = size;
```

Comment:
I think it is possible to support the existing `xs`, `sm`, … `xl` props without changing much code.

```suggestion
    GridAtomicsObj['--Grid-self-column-span'] = props;
    GridAtomicsObj['--Grid-self-width'] = props;
    GridAtomicsObj['--Grid-self-max-width'] = props;
    GridAtomicsObj['--Grid-self-flex'] = props;
```